Donation-platform/src/main/java/com/google/PersonalizedRecommendations.java

```diff
@@ -60,37 +63,45 @@
   // Returns mapping of charities to the scores that determine their rank on the personalzied page
   // (higher score = better match)
   public HashMap<Charity, Double> getCharityScores(List<String> selectedTags) {
+    DbSetUpUtils setUp = new DbSetUpUtils();
+    db = setUp.getDbCalls();
+    //setUp.populateDatabase();                                      //only call once
```

Comment:
remove this line?

Reply:
We decided to leave it commented in there because each person has to run it once locally as we test using the GCP web preview. Also if the database gets erased for some reason, then we have the code available to restore it.

Reply:
A better way to achieve this flow is to make this a function and then take a command line argument to indicate whether running in test mode and if true then run this method.

Reply:
I moved populateDatabase() into a new PersonalizedRecommendations constructor, and it it's in an if statement so that it only gets called if there are no charities in the db.